internal/engine/analytics_reporter.go

```diff
@@ -0,0 +1,81 @@
+package engine
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/windmilleng/wmclient/pkg/analytics"
+
+	"github.com/windmilleng/tilt/internal/store"
+)
+
+// How often to perodically report data for analytics while Tilt is running
+const analyticsReportingInterval = time.Hour * 1
+
+type AnalyticsReporter struct {
+	a       analytics.Analytics
+	store   *store.Store
+	started bool
+}
+
+func (ar *AnalyticsReporter) OnChange(ctx context.Context, st store.RStore) {
+	if ar.started {
+		return
+	}
+
+	state := st.RLockState()
+	defer st.RUnlockState()
+
+	// wait until state has been kinda initialized
+	if !state.TiltStartTime.IsZero() && state.LastTiltfileError() == nil {
+		ar.started = true
+		go func() {
+			for {
+				ar.report()
+				time.Sleep(analyticsReportingInterval)
```

Comment:
we usually do this as

select {
  case <-time.After(interval):
  case <-ctx.Done():
    return
}

so that the goroutine stops when the context is canceled (helpful on tests)